app/classifier/tasks/text/index.cjsx

```diff
@@ -66,6 +66,11 @@ module.exports = React.createClass
     @setState initOffsetHeight: @refs.textInput.offsetHeight
     @updateHeight()
 
+  componentWillReceiveProps: (nextProps) ->
+    if nextProps.task.instruction isnt @props.task.instruction
```

Comment:
I think this works, I'd just check that `nextProps.task isnt @props.task` since two tasks' instructions aren't guaranteed to be unique.